agent/src/util/discovery_configuration_controller.rs

```diff
@@ -204,7 +204,16 @@ async fn delete_instance(
             return Ok(());
         }
         let mut new_instance = instance.clone();
+        // FIXME: temporary workaround for multinode setups.
```

Comment:
```suggestion
        let mut new_instance = Instance {
            metadata: ObjectMeta::default(),
            spec: instance.spec.clone()
        };
        new_instance.spec.nodes = vec![];
```
Only updating the spec is another option here.